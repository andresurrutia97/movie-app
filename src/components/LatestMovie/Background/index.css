.latest-movie__img {
  filter: blur(1px);
  opacity: 0.6;
  margin-top: 64px;
}

@media (min-width: 768px) {
  .latest-movie__img {
    margin-top: 0;
  }
}
