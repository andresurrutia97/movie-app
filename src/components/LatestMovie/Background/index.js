import React from 'react';
import PropTypes from 'prop-types';

import './index.css';
import { getImageUrl } from '../../../utils/urls';

const ImgBackground = ({ img }) => {
  const bgImg = img ? (
    <img className="w-full latest-movie__img" src={getImageUrl(img)} alt="" />
  ) : (
    ''
  );

  return <>{bgImg}</>;
};

ImgBackground.propTypes = {
  img: PropTypes.string,
};

ImgBackground.defaultProps = {
  img: '',
};

export default ImgBackground;
