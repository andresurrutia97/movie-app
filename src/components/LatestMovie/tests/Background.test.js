import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Background from '../Background';

describe('<Background/>', () => {
  const component = <Background img="/imgPath" />;

  it('should render correctly ', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should contain a <img/> element with a image src that includes the path given', () => {
    const wrapper = shallow(component);
    expect(wrapper.find('img')).toHaveLength(1);
    expect(wrapper.find('img').props().src).toMatch(new RegExp('/imgPath'));
  });
});
