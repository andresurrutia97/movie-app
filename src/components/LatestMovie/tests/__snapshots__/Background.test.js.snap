// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Background/> should render correctly  1`] = `
<img
  alt=""
  className="w-full latest-movie__img"
  src="https://image.tmdb.org/t/p/original/imgPath"
/>
`;
